assets/javascripts/discourse/components/upcoming-events-list.gjs

```diff
@@ -0,0 +1,153 @@
+import Component from "@glimmer/component";
+import I18n from "discourse-i18n";
+import { tracked } from "@glimmer/tracking";
+import { inject as service } from "@ember/service";
+import { action } from "@ember/object";
+import { isNotFullDayEvent } from "../lib/guess-best-date-format";
+import ConditionalLoadingSpinner from "discourse/components/conditional-loading-spinner";
+import or from "truth-helpers/helpers/or";
+
+export const DEFAULT_MONTH_FORMAT = "MMMM YYYY";
+export const DEFAULT_DATE_FORMAT = "dddd, MMM D";
+export const DEFAULT_TIME_FORMAT = "LT";
+
+export default class UpcomingEventsList extends Component {
+  @service appEvents;
+  @service siteSettings;
+  @service router;
+  @service postEventApi;
+
+  @tracked isLoading = true;
+  @tracked eventsByMonth = {};
+
+  monthFormat = this.args.params?.monthFormat ?? DEFAULT_MONTH_FORMAT;
+  dateFormat = this.args.params?.dateFormat ?? DEFAULT_DATE_FORMAT;
+  timeFormat = this.args.params?.timeFormat ?? DEFAULT_TIME_FORMAT;
+
+  title = I18n.t("discourse_post_event.upcoming_events_list.title");
+  emptyMessage = I18n.t("discourse_post_event.upcoming_events_list.empty");
+  allDayLabel = I18n.t("discourse_post_event.upcoming_events_list.all_day");
+
+  constructor() {
+    super(...arguments);
+
+    this.appEvents.on("page:changed", this, this.updateEventsByMonth);
+  }
+
+  get shouldRender() {
+    if (!this.categoryId) {
+      return false;
+    }
+
+    const eventSettings =
+      this.siteSettings.events_calendar_categories.split("|");
+
+    return eventSettings.includes(this.categoryId.toString());
+  }
+
+  get categoryId() {
+    return this.router.currentRoute.attributes?.category?.id;
+  }
+
+  get hasEmptyResponse() {
+    return !this.isLoading && Object.keys(this.eventsByMonth).length === 0;
+  }
+
+  @action
+  async updateEventsByMonth() {
+    this.isLoading = true;
+
+    const { events } = await this.postEventApi.categoryEvents(this.categoryId);
```

Comment:
This might need a `try / catch`. What would happen if this request fails? We don't necessarily want to show a modal, but the component might end up in loading state forever?

Reply:
Yeah, I missed that. Maybe something like "Failed to retrieve events. [Try again]" should be enough?

Reply:
Sure, yeah, that makes sense.